hkdf.go

```diff
@@ -151,6 +151,43 @@ func ExtractHKDF(h func() hash.Hash, secret, salt []byte) ([]byte, error) {
 	}
 }
 
+// ExpandHKDFOneShot derives a key from the given hash, key, and optional context info.
+func ExpandHKDFOneShot(h func() hash.Hash, pseudorandomKey, info []byte, keyLength int) ([]byte, error) {
+	if !SupportsHKDF() {
+		return nil, errUnsupportedVersion()
+	}
+
+	md, err := hashFuncToMD(h)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make([]byte, keyLength)
+	switch vMajor {
+	case 1:
+		ctx, err := newHKDFCtx1(md, C.GO_EVP_KDF_HKDF_MODE_EXPAND_ONLY, nil, nil, pseudorandomKey, info)
+		if err != nil {
+			return nil, err
+		}
+		defer C.go_openssl_EVP_PKEY_CTX_free(ctx)
+		if C.go_openssl_EVP_PKEY_derive_wrapper(ctx, base(out), C.size_t(keyLength)).result != 1 {
+			return nil, newOpenSSLError("EVP_PKEY_derive")
+		}
+	case 3:
+		ctx, err := newHKDFCtx3(md, C.GO_EVP_KDF_HKDF_MODE_EXPAND_ONLY, nil, nil, pseudorandomKey, info)
+		if err != nil {
+			return nil, err
+		}
+		defer C.go_openssl_EVP_KDF_CTX_free(ctx)
+		if C.go_openssl_EVP_KDF_derive(ctx, base(out), C.size_t(keyLength), nil) != 1 {
+			return nil, newOpenSSLError("EVP_KDF_derive")
+		}
+	default:
+		panic(errUnsupportedVersion())
```

Comment:
Why `panic` vs. returning the error?

Reply:
This is common practice in this repo. The supported version switches should be exhaustive so we know what to update when we start supporting a new version.

Reply:
What difference does `return nil, errUnsupportedVersion()` vs. `panic(errUnsupportedVersion())` make to the switch's exhaustiveness?

FWIW, most uses of `errUnsupportedVersion` do use `panic`. Only tls1prf.go and hkdf.go return it, so maybe my confusion is in what situations it makes sense to `return` it rather than `panic`. (Maybe a doc comment on `errUnsupportedVersion` would make sense if it's worth explaining it for the future.)

https://github.com/golang-fips/openssl/blob/4f0359e4649dbb346f83b1706519c8a01fc43b59/tls1prf.go#L50-L57

https://github.com/golang-fips/openssl/blob/4f0359e4649dbb346f83b1706519c8a01fc43b59/tls1prf.go#L15-L25

https://github.com/golang-fips/openssl/blob/a47a02278d7e3a14042576f160450e49072eaafc/hkdf.go#L17-L27

https://github.com/golang-fips/openssl/blob/a47a02278d7e3a14042576f160450e49072eaafc/hkdf.go#L113-L115

That said, since this isn't new, going ahead with approval. 😄

Reply:
> What difference does return nil, errUnsupportedVersion() vs. panic(errUnsupportedVersion()) make to the switch's exhaustiveness?

Go doesn't have exhaustive, so it is idiomatic to panic in the `default` case to at least get a runtime panic.

 Good catch about the inconsistency, will fix in a follow up PR.